crates/ntx-builder/src/state/mod.rs

```diff
@@ -59,17 +59,35 @@ pub struct State {
 
     /// A mapping of network note's to their account.
     nullifier_idx: BTreeMap<Nullifier, NetworkAccountPrefix>,
+
+    /// gRPC client used to retrieve the network account state from the store.
+    store: StoreClient,
 }
 
 impl State {
-    pub fn with_committed_notes(notes: impl Iterator<Item = NetworkNote>) -> Self {
-        let mut state = Self::default();
+    /// Load's all available network notes from the store, along with the required account states.
+    pub async fn load(store: StoreClient) -> Result<Self, StoreError> {
+        let mut state = Self {
+            store,
+            accounts: HashMap::default(),
+            queue: VecDeque::default(),
+            in_progress: HashSet::default(),
+            inflight_txs: BTreeMap::default(),
+            nullifier_idx: BTreeMap::default(),
+        };
+
+        let notes = state.store.get_unconsumed_network_notes().await?;
```

Comment:
Do we want to pull in the `store` in here rather than keeping it external? The benefit seems to be small, at least so far. Or do we intend to use it some more in the future?